src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -269,151 +263,156 @@ public boolean chunkExists(int cubeX, int cubeZ) {
 	//==============================
 
 	@Override
-	public boolean cubeExists(int cubeX, int cubeY, int cubeZ) {
-		long columnAddress = AddressTools.getAddress(cubeX, cubeZ);
-		Column column = this.loadedColumns.get(columnAddress);
-		if (column == null || column.unloaded) {
-			return false;
-		}
-		Cube cube = column.getCube(cubeY);
-		return cube != null && !cube.unloaded;
+	public Cube getCube(int cubeX, int cubeY, int cubeZ) {
+		return getCube(new CubeCoords(cubeX, cubeY, cubeZ));
 	}
 
-	public boolean cubeExists(CubeCoords coords) {
-		return this.cubeExists(coords.getCubeX(), coords.getCubeY(), coords.getCubeZ());
+	@Override
+	public Cube getCube(CubeCoords coords) {
+		return getCube(coords, Requirement.GENERATE);
```

Comment:
Like @Barteks2x said, this is a big part of the changes I made... getCube() is now vary strict in its spec, and will NEVER give you null (no more null checks! yay!). In the same way it is more explicit in that you will need to tell getCube() if you need lighting or population to be done.